sktime/proba/base.py

```diff
@@ -158,6 +158,18 @@ def _method_error_msg(self, method="this method", severity="warn", fill_in=None)
         else:
             return msg
 
+    def _get_bc_params(self, to_broadcast, dtype=None):
+        """Fully broadcast parameters of self, given param shapes and index, columns."""
+        length = len(to_broadcast)
+        if hasattr(self, "index") and self.index is not None:
+            to_broadcast += [self.index.to_numpy().reshape(-1, 1)]
+        if hasattr(self, "columns") and self.columns is not None:
+            to_broadcast += [self.columns.to_numpy()]
+        bc = np.broadcast_arrays(*to_broadcast)
+        if dtype is not None:
+            bc = [array.astype(dtype) for array in bc]
+        return bc[:length]
```

Comment:
why are we cutting at the end? just asking for explanation

Reply:
If the `self.index` and/or `self.columns` is appended to `to_broadcast`, then when the child classes implemented `_get_bc_params` they would never them. See here https://github.com/sktime/sktime/blob/main/sktime/proba/t.py#L66 for example.

The end cut off to make sure the broadcasted `self.index` and `self.columns` is never returned.

Reply:
> then when the child classes implemented `_get_bc_params` they would never them.

apologies, that sentence seems incomplete, and I cannot guess a completion, can you kindly rephrase?

Reply:
resolved via explanation